### PR TITLE
Improve overall decryption script

### PR DESCRIPTION
**Pull request by Rem01Gaming**
- Add error handler
  Verify binary from user input is exist.

- Delay 0.07s until it calls execve
  sleep 0.01 isn't enough, bunch scripts and devices combination haven't call execve in that time
  resuling empty decrypt output.

- Add `2>&1` to silence binary output
  Besides completely silence output from binary, this also will fix some binary that stuck
  (can't be killed) while decryption.

### CODE REVIEW

**Comment by kawaii-ghost** [comment]
> Add error handler

No need to add binary handler. Its user responsibility to check if the file exist.

> Delay 0.07s

This is known issue since execution time isn't portable. I'm still configuring this by using `ltrace` and pipe-ing this and that but still didn't work until now. If you can improve it, I will be appreciate it. I'm thinking of using `gdb` but It will be expensive.
```sh
#!/bin/sh
for arg in $@
do
    ltrace -f -e execve $arg > ./tmp &
    tail -f ./tmp | while read line
    do
        if [["$line" == *"execve"*]]; then
            $child = pgrep -f $arg
            kill -STOP $child
            cat /proc/$child/cmdline | sed 's/.*\(#!\)/\1/; $d' > $arg.dec.sh
            kill -TERM $child
        fi
    done
done
```
> Add 2>&1 to silence binary output

I'm only redirecting `stdout` to `/dev/null`, I don't think `stderr` should be also redirected. Could you please elaborate what's binary stuck can't be killed? All signal can be sended not limited to `SIGSTOP` and `SIGTERM` if it's same uid that owning the process.

**Comment by Rem01Gaming** [comment]
> No need to add binary handler. Its user responsibility to check if the file exist.

Yeah but nice to have.

> I'm still configuring this by using ltrace and pipe-ing this and that but still didn't work until now. If you can improve it, I will be appreciate it.

I see some issues from current design from snippets you send :
- Output from Itrace (./tmp) actually empty and not redirect any output, for address this instead using '>' use `tee` with `V=0 2>&1`.
- Instead pipe-ing the `./tmp`, loop while true and `cat ./tmp` until found execve call.
- `pgrep` will give some pid that can't be handled by cat command, this still problematic for current design.
- Since we using `ltrace`, the device should have architecture that match with ELF binary, even the device has supported ABI.

This is my revision from snippets you send
```bash
#!/bin/sh

for arg in $@; do
        ltrace -f -e execve $arg V=0 2>&1 | tee tmp &
        while true; do
                if [[ $(cat ./tmp) == *"execve"* ]]; then
                        child=$(sed -n 1p ./tmp | awk '{print $2}' | sed 's/\]//')
                        #child=$(pidof $arg | head -n 1)
                        echo "Debug: cat cmdline of $child"
                        kill -STOP $child
                        cat /proc/$child/cmdline | sed 's/.*\(#!\)/\1/; $d' >$arg.dec.sh
                        kill -TERM $child
                        break
                fi
        done
done
```
> Could you please elaborate what's binary stuck can't be killed?

Basically the binary just running and can't be killed even with `CTRL + C` and freeze the terminal.

**Comment by kawaii-ghost** [comment]
> `pgrep`...

Either way get the child pid from parent process

> `ltrace` ...

What's the corelation between `ltrace` and ABI?

> Can't be killed

Cool, do you have any idea about Signal IPC on Unix? `CTRL-C` send `SIGINT`, not `SIGTERM` or even `SIGKILL`. The process can ignore the all signal except `SIGKILL`. Seriously, how can redirected file descriptor to `/dev/null` lead to error?

Overall it's all good. That's only cristism from me.

**Comment by Rem01Gaming** [comment]
Sorry for late reply, I've been busy with other projects lately. I will inform you while ready.

Regarding some point I've said before, actually I know what the problem is, I just couldn't explain it well and create misunderstandings.
